src/test/java/io/axoniq/axonserver/connector/impl/AxonServerManagedChannelIntegrationTest.java

```diff
@@ -203,10 +204,11 @@ void connectionRecoveredOnDisconnection() throws Exception {
 
         nextConnectTask.run();
 
-        assertEquals(asList(new ServerAddress("server1"),
-                            new ServerAddress("server2"),
-                            new ServerAddress("server3")),
-                     connectAttempts);
+        // we don't care in which order the connects happened, as long as all hosts have been tried
+        assertEquals(new HashSet<>(asList(new ServerAddress("server1"),
```

Comment:
`Collections.unmodifiableSet(...)` instead of `new HashSet<>(asList(...))`?

Reply:
`Collections.unmodifiableSet(...)` expects a `Set`, so that would change it to `Collections.unmodifiableSet(new HashSet<>(asList(...)`.